From a4423056f59c13171955b5420973f443fb53cb3b Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 07 Oct 2019 11:43:30 +0300
Subject: [PATCH] [SW-0] hl: decrease time before PCIe link is established

1) move preboot_pcie_init() upwards
2) remove log before link created (saves 2mS)

TODO: Need to move all code we can from preboot to SPL this will
decrease image size and will make preboot copy time faster. For example
move preboot_version() and also hl_pll_init()
Also we may add validation for image size in Makefile

Change-Id: Ic05d4b52babbd3794876763fec0ec59e9bb9c04c
---

diff --git a/common/preboot.c b/common/preboot.c
index 44b3b21..9279d6a 100644
--- a/common/preboot.c
+++ b/common/preboot.c
@@ -94,7 +94,7 @@
 		return;
 	}
 
-	preboot_puts("Waiting for PCIe link...\n");
+	//preboot_puts("Waiting for PCIe link...\n");
 	preboot_pcie_wait_for_link();
 	preboot_puts("PCIe link is UP!\n");
 }
@@ -163,8 +163,8 @@
 	hl_jtag_init();
 	preboot_sync_btl();
 	hl_uart_init();
-	preboot_version();
 	preboot_pcie_init();
+	preboot_version();
 	preboot_delay();
 	hl_pll_init();
 	hl_pll_wa();
